### PR TITLE
fix: Toast is removed using onClick

### LINKED ISSUE #9204: Unable to remove the toast using Onclick
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Unable to remove the toast using Onclick.

https://github.com/calcom/cal.com/assets/81948346/c00f2d27-73b6-4649-b7de-9fcc387f57d1

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?
Before it takes time to hide the toast and there was no click event to hide it. Now user can hide the toast.

https://github.com/calcom/cal.com/assets/81948346/d496ad3d-e5c7-4b87-81fa-8bd5264bceb2

Fixes #9204 

## Type of change

- New feature (non-breaking change which adds functionality)
## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by sean-brydon — COMMENTED** [review]
Not sure on a11y here - should this div become a button if we attach an onclick to perserve the default a11y that comes with a button

File: packages/ui/components/toast/showToast.tsx [hunk 1]
```diff
@@ -6,53 +6,58 @@ import { Check, Info } from "../icon";
...
     <span>
       <Check className="h-4 w-4" />
     </span>
     <p data-testid="toast-success">{message}</p>
   </div>
 );
 
-export const ErrorToast = ({ message, toastVisible }: IToast) => (
+export const ErrorToast = ({ message, toastVisible, onClose }: IToast) => (
   <div
```
Comment (PeerRich) [inline packages/ui/components/toast/showToast.tsx:27]:
```suggestion
  <button
```

File: packages/ui/components/toast/showToast.tsx [hunk 2]
```diff
@@ -6,53 +6,58 @@ import { Check, Info } from "../icon";
...
-      toastVisible && "animate-fade-in-up"
-    )}>
+      toastVisible && "animate-fade-in-up cursor-pointer"
+    )}
+    onClick={onClose}>
     <span>
       <Info className="h-4 w-4" />
     </span>
     <p data-testid="toast-error">{message}</p>
   </div>
```
Comment (PeerRich) [inline packages/ui/components/toast/showToast.tsx:37]:
```suggestion
  </button>
```

File: packages/ui/components/toast/showToast.tsx [hunk 3]
```diff
@@ -6,53 +6,58 @@ import { Check, Info } from "../icon";
...
-      toastVisible && "animate-fade-in-up"
-    )}>
+      toastVisible && "animate-fade-in-up cursor-pointer"
+    )}
+    onClick={onClose}>
     <span>
       <Info className="h-4 w-4" />
     </span>
     <p data-testid="toast-warning">{message}</p>
   </div>
```
Comment (PeerRich) [inline packages/ui/components/toast/showToast.tsx:51]:
```suggestion
  </button>
```

File: packages/ui/components/toast/showToast.tsx [hunk 4]
```diff
@@ -6,53 +6,58 @@ import { Check, Info } from "../icon";
...
 );
 
-export const DefaultToast = ({ message, toastVisible }: IToast) => (
+export const DefaultToast = ({ message, toastVisible, onClose }: IToast) => (
   <div
```
Comment (PeerRich) [inline packages/ui/components/toast/showToast.tsx:55]:
```suggestion
  <button
```

**Comment by PeerRich** [comment]
> Not sure on a11y here - should this div become a button if we attach an onclick to perserve the default a11y that comes with a button

agree. done

**Review by PeerRich — DISMISSED** [review]

**Comment by anikdhabal** [comment]
> Not sure on a11y here - should this div become a button if we attach an onclick to perserve the default a11y that comes with a button

undrstood, thanks

**Comment by anikdhabal** [comment]
> > Not sure on a11y here - should this div become a button if we attach an onclick to perserve the default a11y that comes with a button
> 
> agree. done

Thank you sir

**Review by PeerRich — APPROVED** [review]

**Pull request merged**